src/FeatureDiscovery.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+class FeatureDiscovery extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {mounted: false};
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+    this.close = this.close.bind(this);
+    this.reference = this.reference.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({mounted: true});
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.props.open && nextProps.open) {
+      this.open();
+    } else if (nextProps.open === false) {
+      this.close();
+    }
+  }
+
+  reference(element) {
+    this._element = element;
+    if (this.props.open) {
+      this.open();
+    }
+  }
+
+  open() {
+    const element = $(this._element);
+    element.tapTarget('open');
+
+    const parent = element.parent('.tap-target-wrapper');
+    const origin = parent.find('.tap-target-origin');
+    const target = $('#' + this.props.target);
+    origin.click(
+      () => {
+        target.trigger('click')
+        target.trigger('mouseover')
+      }
+    )
```

Comment:
Could you explain a bit what is happening here? reading the materialize-css docs it seems that you only need `element.tapTarget('open');` to show the featureDiscovery?

Reply:
This is here because when the _tap target_ is opened, a new button is created on top of the existing one
and no events are delegated to the original button. Which is essentially what we have there.